src/make_trade.ts

```diff
@@ -188,6 +210,34 @@ async function giveAllowanceIfNecessary(
   const allowance = await erc20.allowance(trader.address, allowanceManager);
   if (allowance.lt(sellAmount)) {
     await erc20.connect(trader).approve(allowanceManager, MAX_ALLOWANCE);
-    console.log(`Successfully set allowance for ${sellToken.name}`);
+    console.log(`✅ Successfully set allowance for ${sellToken.name}`);
   }
 }
+
+async function waitForTrade(
+  contract: string,
+  trader: string,
+  uid: string,
+  ethers: HardhatEthersHelpers
+): Promise<boolean> {
+  const filter = {
+    address: contract,
+    topics: [
+      keccak("Trade(address,address,address,uint256,uint256,uint256,bytes)"),
+      hexZeroPad(trader, 32),
+    ],
+  };
```

Comment:
nit: I think it would be nicer to use the generated Ethers filter methods:
```ts
const filter = settlement.filters.Trade(trader);
```

Reply:
Feel free to ignore this if getting a contract instance is not worth the effort.